Feature/party detail

YusukeUenaka opened the pull request:
conflictを解消し、developをマージしたものをコミットしましたので、
developにマージお願いします。

YusukeUenaka committed:
ダイアログ表示。TableViewのパーツ配置。

YusukeUenaka committed:
ファイル追加

YusukeUenaka committed:
・スイッチを押した時、セルの背景色変更処理を追記
・スイッチを押した時、Realmを更新する処理を追記
・通知ボタンの枠を丸角枠に変更
・アイコンを指定画像に追記

YusukeUenaka committed:
スイッチボタン押下でフラグとセルの背景、集計結果の表示を変更する処理を追記。

YusukeUenaka committed:
conflictを解消し、テストデータを一部変更

YusukeUenaka committed:
conflictを解消し、テストデータを修正

takuto1212 commented:
#4のプルリクがレビュー完了しないままマージしているようです。今リバートするのは難しいと思うので、こちらプルリクで修正お願いします。

1. 全体的にコメントがきちんと書けていません。
以下のような<#〜#>の部分を適切に書き直してください。
`/// メンバー登録
 ///
 /// - Parameters:
 ///   - partyId: <#partyId description#>
 ///   - memberName: <#memberName description#>
 ///   - mailAddress: <#mailAddress description#>
 ///   - paymentCompleteFlag: <#paymentCompleteFlag description#>
 ///   - paymentAmount: <#paymentAmount description#>
 /// - Returns: <#return value description#>
 public static func entryMember(partyId: Int, memberName: String, mailAddress: String, paymentCompleteFlag: Int, paymentAmount: Int) -> Bool {`

2. Member.swift
コロンの前のスペースは不要です。
`×　dynamic var partyId : Int = 0 //飲み会ID
○　dynamic var partyId: Int = 0 //飲み会ID
`

3.protocol cellaction
一旦CustomMemberCellDelegateにしましょう。
cellactionだとデリゲートであることが分かりません。
あと大文字始まりで。

4.PartyDetailViewController.swift
`var party = DBManager.searchParty(partyId: pId)
let totalAmount: Int = party[0].totalAmount`
これだと結果が0件のときにクラッシュします。
` if party.isEmpty {
    return
}`
を挟んで安全に処理してあげましょう。
※本来であればここでエラーを出すなど適切な処理をしてあげるべきです。

5.強制アンラップの禁止
members! のように強制アンラップするのはやめましょう。
イニシャライザの中で初期化をすればオプショナル型にする必要がなくなります。
イニシャライザの書き方はPartyAddViewControllerを参考にしてください。

6. Member.paymentCompleteFlag
０、１で判定していますが、Bool型にすれば良いのでは？


7.PartyDetailViewController.swift
`extension PartyDetailViewController: cellaction {
    func changeCellColor(swisOn: Bool, memberCell: UIView) {
        if swisOn == true {`
swisOnはBool型なので、`if swisOn {`でよいのでは？

8.warningが4箇所出ています。
全て解消してからpushプルリク出してください。

他にも細かいところはたくさんありますが、一旦ここまで。

YusukeUenaka committed:
・プルリクの指摘内容を反映（1〜７）
・No.8：PartyDetailViewControllerのchangeCellColorのワーニングは解消方法がわかりません。
editButtonのワーニングは、画面遷移の接続時に解消されるため、対応していません。
・複数メンバーが存在する時の、switchの挙動がおかしかったため、修正しました。

takuto1212 commented:
前回1の指摘がDBManager.swiftしか対応していないようです。
他のファイルについても対応お願いします。

takuto1212 commented:
「s30.png」ですが、ファイル直置きしているようですが、
Assets.xcassetsを使うようにしましょう。

YusukeUenaka committed:
レビュー指摘対応

YusukeUenaka commented:
西村さんと対面で指摘対応
